internal/nexodus/utils_darwin.go

```diff
@@ -3,15 +3,49 @@
 package nexodus
 
 import (
+	"bytes"
 	"fmt"
-	"github.com/vishvananda/netlink"
-	"go.uber.org/zap"
 	"net"
+	"os"
+	"os/exec"
 	"strings"
+
+	"github.com/vishvananda/netlink"
+	"go.uber.org/zap"
 )
 
-// RouteExistsOS currently only used for darwin build purposes
-func RouteExistsOS(s string) (bool, error) {
+// RouteExistsOS checks to see if a route exists for the specified prefix
+func RouteExistsOS(prefix string) (bool, error) {
+	if err := ValidateIp(prefix); err != nil {
+		return false, err
+	}
+
+	r, w, err := os.Pipe()
+	if err != nil {
+		return false, err
+	}
+	defer r.Close()
+	ns := exec.Command("netstat", "-r", "-n")
+	ns.Stdout = w
+	if err = ns.Start(); err != nil {
+		return false, err
+	}
+	defer func() {
+		_ = ns.Wait()
+	}()
+	w.Close()
```

Comment:
If ns.Start() fails, the write end of the pipe is never closed. I see the blog post I sent you has the same thing. It looks like a bug to me. What do you think? I would move `w.Close()` to be `defer w.Close()` right after `defer r.Close()`.